php/libraries/Database.class.inc

```diff
@@ -749,17 +749,27 @@ class Database
     /**
      * Runs a query as a prepared statement and returns the first row as an
      * associative array. Automatically adds a limit clause to the query being
-     * run for efficiency.
+     * run for efficiency. Limit clause can be disabled by setting false as
+     * the third argument.
      *
-     * @param string $query  The SQL SELECT query to be run
-     * @param array  $params Values to use for binding to prepared statement
+     * @param string $query           The SQL SELECT query to be run
+     * @param array  $params          Values to use for binding to prepared
+     *                                statement
+     * @param bool   $efficiencyLimit Determines whether to use LIMIT clause or not
      *
      * @return ?array Associative array of form ColumnName => Value for each column
      *                in the first row of the query
      */
-    function pselectRow(string $query, array $params) : ?array
-    {
-        $rows = $this->pselect($query . " LIMIT 2", $params);
+    function pselectRow(
+        string $query,
+        array $params,
+        bool $efficiencyLimit = true
+    ) : ?array {
+
+        if ($efficiencyLimit) {
+            $query .= " LIMIT 2";
+        }
+        $rows = $this->pselect($query, $params);
```

Comment:
pselectRow select only one row (LIMIT *1*) at the end of the query. Do not know why it was LIMIT 2. This might be the source cause of your problem.

Reply:
It's supposed to be 2 in order to trigger the error. If it returns 2, we say "you used this wrong".

Reply:
@johnsaigle  need to update my VM, it as `LIMIT 1`.  and no code for exception.

Thanks for the info